monitoring/tests/test_finished_build_stats_publisher.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+import unittest
+from unittest.mock import MagicMock
+from finished_build_stats_publisher import FinishedBuildStatsPublisher
+from collector import ICollector
+from publisher import IPublisher
+from stats import BuildStats
+
+
+class TestFinishedBuildStatsPublisher(unittest.TestCase):
+    class CollectSideEffect(object):
```

Comment:
Maybe definition of this class should be inside `test_with_retries`.

Maybe it should be a generator, like:
```
>>> def c():
...     for i in range(3):
...         yield i
...     return
```